### ISSUE TITLE
[BUG] Scroll problem

### ISSUE THREAD

**Issue by liangge013** [tneotia/html-editor-enhanced#114]
When I use ToolbarPosition.belowEditor, I click on the page to start editing, and it cannot locate the line where the cursor is. It looks like it is blocked by the toolbar. 

https://user-images.githubusercontent.com/11863286/129859525-b9e87f9a-2733-4475-90a2-f9e4ea5deee1.MP4

**Comment by tneotia** [comment]
Unfortunately I don't think there is a way to make this cursor scroll into view but I can check and see if there is any solution. Whenever the user starts typing it should scroll itself into view though.

**Comment by liangge013** [comment]
Thanks for your reply.
Why does the page scrolling look like a conflict occurs whenever I edit?

https://user-images.githubusercontent.com/11863286/129994986-85cb75ad-ba46-41c1-b110-244f0f87b010.mov

**Comment by tneotia** [comment]
That is odd.... What is the widget code for your page? Is your page only the html editor or are there other components on the page?

**Comment by liangge013** [comment]
`Widget build(BuildContext context) {
  // TODO: implement build

  return Scaffold(
      backgroundColor: Color(0xffeeeeee),
      appBar: AppBar(
        brightness: Brightness.light,
        centerTitle: true,
        elevation: 0,
        title: Text('Note',style: TextStyle(color: Colors.black),),
        backgroundColor: Color(0xffffffff),
        iconTheme: IconThemeData(color: Color(0xff666666)),
        actions: <Widget>[
          FlatButton(
              disabledTextColor: Color(0xff999999),
              textColor: Color(0xff4a90e2),
              child: Text('save',style: TextStyle(fontSize: 18)),
              onPressed:() async{
                final txt = await Econtroller.getText();
                String documentsPath = (await getApplicationDocumentsDirectory()).path;
                String path = documentsPath+'/'+widget.noteModel.noteHashId+'.html';
                final file = new File(path);
                if(!file.existsSync()) {
                  await new Directory(dirname(path)).create(recursive: true).then((value){
                    final file = File(path);
                    file.writeAsString(txt).then((value){
                      Navigator.of(context).pop();
                    });
                  });
                }else{
                  file.writeAsString(txt).then((value){
                    Navigator.of(context).pop();
                  });
                }
              }
          )
        ],
      ),
      body: Container(
          color: Colors.white,
          child:
          HtmlEditor(
            controller: Econtroller,
            callbacks: Callbacks(
              onFocus: () {
                setState(() {
                  isEdit = true;
                });
              },
              onBlur: () {
                setState(() {
                  isEdit = false;
                  secondToolBar = false;
                });
              },
              onChangeSelection: (EditorSettings settings) {
                print('111111111111111111111111111111111');
                setState(() {
                  fontColor = settings.foregroundColor;
                });
              },
            ),
            //required
            htmlEditorOptions: HtmlEditorOptions(
                hint: "",
                initialText: content,
                autoAdjustHeight: false,
                adjustHeightForKeyboard: false),
            otherOptions: OtherOptions(
              height: MediaQuery.of(context).size.height - 120,
            ),
            htmlToolbarOptions: HtmlToolbarOptions(
              toolbarPosition: ToolbarPosition.aboveEditor,
              toolbarItemHeight: 50,
              renderSeparatorWidget: false,
              defaultToolbarButtons: [
                InsertButtons(
                    link: false,
                    video: false,
                    audio: false,
                    table: false,
                    hr: false,
                    otherFile: false),
                FontButtons(
                    clearAll: false,
                    superscript: false,
                    subscript: false),
                ColorButtons(),
                StyleButtons(),
                FontSettingButtons(fontName: false, fontSizeUnit: false),
                ListButtons(listStyles: false),
                ParagraphButtons(
                    textDirection: false,
                    lineHeight: false,
                    caseConverter: false),
              ],
              onButtonPressed: (ButtonType type, bool status,
                  Function() updateStatus) {
                if (type == ButtonType.picture) {
                  showImageDialog(
                      context); //show your custom dialog here or directly open image picker
                  return false; //this tells the plugin we handled image in a custom way
                } else if (type == ButtonType.foregroundColor) {
                  return false;
                }
                return true; //this tells the plugin to handle everything else as normal
              },
            ),
          )
      )
  );
}`

**Comment by tneotia** [comment]
Ohh it is because of these options:

```
autoAdjustHeight: false,
                adjustHeightForKeyboard: false)
```
Best experience will be to make these true to remove the double scroll.

**Comment by liangge013** [comment]
I set them all to true, the problem still exists. 
And if I set them to true,  use Econtroller.setFullScreen() will cause the page to fail to slide

**Comment by tneotia** [comment]
Can you send an updated screen recording of the new behavior? Just curious what it is doing now.

**Comment by liangge013** [comment]
autoAdjustHeight: true,
adjustHeightForKeyboard: true


https://user-images.githubusercontent.com/11863286/130009441-0a822af7-650d-4f68-b516-417b487a74e8.mp4

**Comment by tneotia** [comment]
Hmm something is still off, I will test your page code myself later today and see why it is doing that.

**Comment by tneotia** [comment]
Thank you for the report, sorry it took so long, it is fixed now :)

**Comment by reke592** [comment]
Hi sir, I would like to report same issue in web

```dart
              HtmlEditor(
                controller: widget.controller,
                htmlEditorOptions: HtmlEditorOptions(
                  hint: 'Type your text here...',
                  shouldEnsureVisible: false,
                  autoAdjustHeight: true,
                  adjustHeightForKeyboard: true
                  //initialText: "<p>text content initial, if any</p>",
                ),
                ...
```
![htmleditoradv](https://user-images.githubusercontent.com/17094894/134490929-fa2dffd4-2605-4c2d-83c7-0fc2d6329f8a.gif)

**Comment by liangge013** [comment]
It looks like it hasn't been fixed.

**Comment by tneotia** [comment]
Fixed in mobile, wasn't aware it was an issue in web. @reke592 can you send me your full code for that page for my testing?

**Comment by reke592** [comment]
Apologies for the late reply, below is the sample code

```dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:html_editor_enhanced/html_editor.dart';

class TestForm extends StatefulWidget {
  final String? id;
  final bool isNew;
  final bool isEdit;

  TestForm({this.id, this.isNew = false, this.isEdit = false});

  @override
  State<StatefulWidget> createState() {
    return _TestForm();
  }
}

class _TestForm extends State<TestForm> {
  final HtmlEditorController _descriptionsController = HtmlEditorController();
  final HtmlEditorController _qualificationsController = HtmlEditorController();
  String _editorInitialDescription = '';
  String _editorInitialQualifications = '';

  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Row(
      children: [
        Expanded(
          flex: 3,
          child: Column(
            children: [
              Expanded(
                child: content(),
              ),
            ],
          ),
        ),
        VerticalDivider(),
        Expanded(
          flex: 1,
          child: sidePanel(),
        )
      ],
    );
  }

  Widget content() {
    return SingleChildScrollView(
      // controller: _scrollController,
      child: Container(
        padding: EdgeInsets.all(20),
        child: Column(
          mainAxisSize: MainAxisSize.max,
          children: [
            Card(child: textArea1()),
            Card(child: textArea2()),
          ],
        ),
      ),
    );
  }

  Widget sidePanel() {
    return SingleChildScrollView(
      child: Container(
        padding: EdgeInsets.all(20),
        // color: Colors.blue,
        child: Column(
          children: [
            Text('test1'),
            Text('test2'),
          ],
        ),
      ),
    );
  }

  Widget textArea1() {
    return Container(
      padding: EdgeInsets.all(20),
      child: HtmlEditor(
        controller: _descriptionsController,
        htmlToolbarOptions: HtmlToolbarOptions(
            toolbarPosition: ToolbarPosition.custom,
            toolbarType: ToolbarType.nativeGrid),
        htmlEditorOptions: HtmlEditorOptions(
            hint: 'Job description...',
            shouldEnsureVisible: false,
            autoAdjustHeight: true,
            adjustHeightForKeyboard: true,
            initialText: _editorInitialDescription),
        otherOptions: OtherOptions(
          height: MediaQuery.of(context).size.height / 2,
        ),
      ),
    );
  }

  Widget textArea2() {
    return Container(
      padding: EdgeInsets.all(20),
      child: HtmlEditor(
        controller: _qualificationsController,
        htmlToolbarOptions: HtmlToolbarOptions(
            toolbarPosition: ToolbarPosition.custom,
            toolbarType: ToolbarType.nativeGrid),
        htmlEditorOptions: HtmlEditorOptions(
            hint: 'Qualifications...',
            shouldEnsureVisible: false,
            autoAdjustHeight: false,
            adjustHeightForKeyboard: false,
            initialText: _editorInitialQualifications),
        otherOptions: OtherOptions(
          height: MediaQuery.of(context).size.height / 2,
        ),
      ),
    );
  }
}
```

**Comment by liangge013** [comment]
The problem of sliding conflict was solved, but new problems appeared.When I click on a row to start editing, the page cannot be positioned above the keyboard

**Comment by tneotia** [comment]
> The problem of sliding conflict was solved, but new problems appeared.When I click on a row to start editing, the page cannot be positioned above the keyboard

Can you open a separate issue for this one please :)

**Comment by tneotia** [comment]
@reke592 fixed via e91339c2f5e827073510504ae7b01a6d8079c281
